docs/sources/index.md

```diff
@@ -294,4 +294,37 @@ TypesafeConfig.fromHoconString(jsonString, automaticDescription)
 
 ```
 
+## Commandline Source
+Simply source config from command line arguments
+```scala mdoc:silent
+final case class A(key1: String, key2: String)
+
+val cmdLineArgs = "--key1 value1 --key2 value2"
+val source = ConfigSource.fromCommandLineArgs(cmdLineArgs.split(' ').toList)
+```
+With nested path in command line arguments
+```scala mdoc:silent
+final case class SparkConf(key1: String, key2: String)
+final case class A(conf: SparkConf, key3: String)
+
+val cmdLineArgs = "--conf.key1 v1 --conf.key2 v2 --key3 v3"
+val source = ConfigSource.fromCommandLineArgs(cmdLineArgs.split(' ').toList, keyDelimiter = Some('.'))
+```
+The config with a complex source of command line arguments is what every production service needs
+```scala mdoc:silent
+final case class UserPassword(k2: String, k3: String)
+final case class DatabaseConfig(conf: UserPassword, url: String)
+final case class VaultConfig(userPassword: UserPassword)
+final case class SparkConfig(databaseConfig: DatabaseConfig, numberOfExecutors: Int)
+final case class AppConfig(sparkConfig: SparkConfig, vault: VaultConfig, users: List[String], region: List[String])
+
+val argss = "--conf -database.username=1 --conf -database.password=hi --conf.database.url=jdbc://xyz --conf -num_execs=10 --vault.username=3 --vault.password=10 --vault.something=11 --users 1 --users 2 --region 111,112"
```

Comment:
In this example let’s talk about only one type of nesting and that is nesting using dots. 

That’s is talk about only —conf.database.url.....  
and not —conf database.url

This will create less confusions amongst users. 


We can add another example that talks about “hey you could also do —conf -num_executors=1   
—conf -num_tasks=2 —conf -num_mappers=10 , where you can consider conf as a way to nest. 
for example, either retrieve the map using map(conf)(string) retrieving all the key value pairs under spark ”conf”. Or even similar case classes as above. 

As a third example, we can talk about lists

—url bla.com —url xyz.com —url abc.com

list(url)(string) returns all the urls 

This is really useful for many users. Especially, it is useful when users really run of an actual delimiter because their url is complex enough with :: and ; and ,  etc.

And lets mention we can also do 

—url bla.com,xyz.com,abc.com 
And pass a value delimiter to get them as list using the same description.

Reply:
We can restrict some functionalities depending on the feedback. As of now, let’s don’t mix all the patterns in 1 example. Instead, as I suggested in the above example, we can have multiple small examples

Reply:
Also I think we did this command line a day. We can give more thoughts on to this going forward. Let’s  say “experimental” in the document similar to what we did with hocon in the beginning.  


Also let’s explicitly mention x=y doesn’t work because x doesn’t start with - . I have seen both patterns, so to be honest I don’t know if it’s right or wrong to have that restriction.